packages/dx-react-scheduler/src/plugins/day-view.tsx

```diff
@@ -81,151 +51,56 @@ class DayViewBase extends React.PureComponent<VerticalViewProps, ViewState> {
     timeTableRowComponent: 'TimeTableRow',
   };
 
-  scrollingStrategyComputed = memoize((viewName, scrollingStrategy) => getters =>
-    computed(getters, viewName!, () => scrollingStrategy, getters.scrollingStrategy));
-
-  timeTableElementsMetaComputed = memoize((viewName, timeTableElementsMeta) => getters =>
-    computed(getters, viewName!, () => timeTableElementsMeta, getters.timeTableElementsMeta));
-
-  viewCellsDataComputed = memoize((viewName, startDayHour, endDayHour, cellDuration) => getters =>
-    computed(
-      getters,
-      viewName,
-      viewCellsDataBaseComputed(startDayHour, endDayHour, cellDuration), getters.viewCellsData,
-    ));
-
-  cellDurationComputed = memoize((viewName, cellDuration) => getters =>
-    computed(getters, viewName!, () => cellDuration, getters.cellDuration));
-
-  intervalCountComputed = memoize((viewName, intervalCount) => getters =>
-    computed(getters, viewName!, () => intervalCount, getters.intervalCount));
-
-  availableViews = memoize((viewName, displayName) => ({ availableViews }) =>
-    availableViewsCore(availableViews, viewName, displayName));
-
-  currentView = memoize((viewName, viewDisplayName) => ({ currentView }) => (
-    currentView && currentView.name !== viewName
-      ? currentView
-      : { name: viewName, type: TYPE, displayName: viewDisplayName }
-  ));
-
-  endViewDateComputed: ComputedFn = (getters) => {
-    const { name: viewName } = this.props;
-    return computed(
-      getters, viewName!, endViewDateBaseComputed, getters.endViewDate,
-    );
-  }
-
-  startViewDateComputed: ComputedFn = (getters) => {
-    const { name: viewName } = this.props;
-    return computed(
-      getters, viewName!, startViewDateBaseComputed, getters.startViewDate,
-    );
-  }
-
-  updateRects = memoize((
-    appointments, startViewDate, endViewDate, viewCellsData, cellDuration, currentDate,
-  ) => (cellElementsMeta) => {
-    const rects = verticalTimeTableRects(
-      appointments, startViewDate, endViewDate, [],
-      viewCellsData, cellDuration, cellElementsMeta,
-    );
-
-    this.setState({ rects, timeTableElementsMeta: cellElementsMeta });
-  });
-
-  setScrollingStrategy = (scrollingStrategy) => {
-    this.setState({ scrollingStrategy });
-  }
-
   render() {
     const {
-      layoutComponent: Layout,
+      layoutComponent,
```

Comment:
We can spread many of this props to the BasicView

Reply:
I don't like using the spread operator because it's not clear to understand what properties have passed into the component. Also `base-view.types` have not all required properties.